crates/telio-traversal/src/batcher.rs

```diff
@@ -242,23 +284,371 @@ where
+                            }
+
+                            }
+                            Ok(ac) = batcher.get_actions(last_activity) => {
+                                for a in ac {
+                                    a.1(&mut checker).await.unwrap();
+                                }
+                            }
+                        }
+                    }
+                });
+            }
+
+            // begin the actual test
+            spawn_batcher_poller(
+                params.trigger_enabled,
+                left_recv,
+                batcher_left,
+                test_checker_left.clone(),
+            );
+
+            // add another action after a delay to the peer so they would be misaligned
+            sleep(params.delay_to_add_second_peer).await;
+
+            spawn_batcher_poller(
+                params.trigger_enabled,
+                right_recv,
+                batcher_right,
+                test_checker_right.clone(),
+            );
+
+            sleep(params.test_duration).await;
+
+            let left_values = test_checker_left
+                .lock()
+                .await
+                .values
+                .iter()
+                .map(|v| v.1.elapsed().as_secs() as i64)
+                .collect::<Vec<i64>>();
+            let right_values = test_checker_right
+                .lock()
+                .await
+                .values
+                .iter()
+                .map(|v| v.1.elapsed().as_secs() as i64)
+                .collect::<Vec<i64>>();
+
+            // Produces an average diff between data points. It compares points by finding
+            // the closest value and compares against it. Produces an average of differences. The closer
+            // to 0 the more similar the datasets
+            fn average_alignment(left: &[i64], right: &[i64]) -> f64 {
+                if left.is_empty() || right.is_empty() {
+                    panic!("no data present");
+                }
+
+                fn find_closest_value(v: i64, data: &[i64]) -> i64 {
+                    data.iter()
+                        .min_by_key(|&&ts| (ts - v).abs())
+                        .copied()
+                        .unwrap_or(v)
+                }
+
+                let mut total_diff = 0.0;
+                let mut count = 0;
+
+                for &t1 in left {
+                    let closest_t2 = find_closest_value(t1, right);
+                    total_diff += (t1 - closest_t2).abs() as f64;
+                    count += 1;
+                }
+
+                total_diff / count as f64
+            }
+
+            fn average_difference(numbers: &[i64]) -> f64 {
+                if numbers.len() < 2 {
+                    panic!("not enough elements");
+                }
+
+                let total_diff: i64 = numbers
+                    .iter()
+                    .zip(numbers.iter().skip(1))
+                    .map(|(a, b)| a - b)
+                    .sum();
```

Comment:
```suggestion
                let total_diff: i64 = numbers
                    .windows(2)
                    .map(|w| w[0] - w[1])
                    .sum();
```

Reply:
A bit of a nit

Reply:
Sorry but I am Linux user /s

Fixed that :)